src/stress.jl

```diff
@@ -50,35 +50,35 @@ Reference:
 """
 module Stress
 
-using GeometryTypes, Compat, FixedSizeArrays
+using GeometryTypes, Compat, StaticArrays
 import Base: start, next, done, *
 
-function (*){T<:LinAlg.BlasFloat,S<:FixedArray}(A::StridedMatrix{T}, x::StridedVector{S})
+function (*){T<:LinAlg.BlasFloat,S<:StaticArray}(A::StridedMatrix{T}, x::StridedVector{S})
     A_mul_B!(similar(x, S, size(A,1)), A, x)
 end
 
-immutable Layout{M1<:AbstractMatrix, M2<:AbstractMatrix, VP<:AbstractVector,FT<:AbstractFloat}
+immutable Layout{M1<:AbstractMatrix, M2<:AbstractMatrix, VP<:AbstractVector, FT<:AbstractFloat}
     δ::M1
     weights::M2
     positions::VP
-    pinvLw::Matrix{FT}
+    pinvLw::Matrix{Float64}
     iterations::Int
     abstols::FT
     reltols::FT
     abstolx::FT
 end
 
 
-function initialweights(D, T=eltype(D))
+function initialweights(D)
     map(D) do d
-        x = T(d^(-2.0))
-        isfinite(x) ? x : zero(T)
+        x = Float64(d^(-2.0))
```

Comment:
Not the eltype, map function is generating different outputs in Julia 0.5 and 0.6
In 0.5, it generates sparse matrix with Float elements : 
```
julia> function fx(a,t=eltype(a))
              map(a) do d
              x=t(d^(-2.0))
              isfinite(x)?x:zero(t)
              end
              end
fx (generic function with 2 methods)

julia> fx(a,Float32)
30×30 sparse matrix with 116 Float32 nonzero entries:
	[2 ,  1]  =  1.0
	[3 ,  1]  =  1.0
	[4 ,  1]  =  1.0
	[5 ,  1]  =  1.0
	[6 ,  1]  =  1.0
	[7 ,  1]  =  1.0
	[8 ,  1]  =  1.0
```

In 0.6 it is generating matrix with Any elements : 
```
julia> function fx(a,t=eltype(a))
              map(a) do d
              x=t(d^(-2.0))
              isfinite(x)?x:zero(t)
              end
              end
fx (generic function with 2 methods)

julia> fx(a,Float32)
30×30 SparseMatrixCSC{Any,Int64} with 116 stored entries:
  [2 ,  1]  =  1.0
  [3 ,  1]  =  1.0
  [4 ,  1]  =  1.0
  [5 ,  1]  =  1.0
  [6 ,  1]  =  1.0
  [7 ,  1]  =  1.0
  [8 ,  1]  =  1.0
```
What do you suggest?